go-controller/pkg/ovn/ovn.go

```diff
@@ -444,7 +450,7 @@ func networkStatusAnnotationsChanged(oldPod, newPod *kapi.Pod) bool {
 func (oc *Controller) ensurePod(oldPod, pod *kapi.Pod, addPort bool) error {
 	// Try unscheduled pods later
 	if !util.PodScheduled(pod) {
-		return fmt.Errorf("failed to ensurePod %s/%s since it is not yet scheduled", pod.Namespace, pod.Name)
```

Comment:
we'd want to retry creating the pod later and iiuc it relies on an error being returned

Reply:
why do we rely on this error? the Pod object is going to be updated again once the pod is scheduled...

Reply:
also before we used to return false which is why this was converted to return error...maybe like casey points out its not necessary..

Reply:
yeah I would just return nil and let the pod update send us a new event

Reply:
tried using nil, kind cluster failed to come up. lots of 
```
FailedCreatePodSandBox   pod/coredns-64897985d-h2jj7   Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "58a2db152ee352899b241593ad05c75daf4b407506c69095a846d9d6b5010e14": CNI request failed with status 400: '[kube-system/coredns-64897985d-h2jj7 58a2db152ee352899b241593ad05c75daf4b407506c69095a846d9d6b5010e14] [kube-system/coredns-64897985d-h2jj7 58a2db152ee352899b241593ad05c75daf4b407506c69095a846d9d6b5010e14] failed to get pod annotation: timed out waiting for annotations: context deadline exceeded
```

I think what happens is:
the pod is unscheduled -> we treat the create as success and delete it from retry queue -> pod gets updated -> `objectsToRetry.checkRetryObj(newKey)` returns false (this param tells ensurePod if it should create a logicalport) = pod never gets a logical port allocated.

Reply:
right that `addPort` logic is tricky probably explains why we treated an unscheduled pod as error until now.
We use that flag to determine if  its a new addLSP or not I guess (although addLSP function does have the logic in place to determine if its an existing LSP or update), probably to avoid extra computations?

Reply:
yeah I *think* that was the reason (extra txns and unnecessary code execution), but honestly it took so long to get that code right I wouldn't touch it. Let's just change the update case to be like:

```
	case factory.PodType:
		oldPod := oldObj.(*kapi.Pod)
		newPod := newObj.(*kapi.Pod)
		newKey, err := getResourceKey(objectsToRetry.oType, newObj)
		if err != nil {
			return err
		}
		return oc.ensurePod(oldPod, newPod, objectsToRetry.checkRetryObj(newKey) || util.PodScheduled(oldPod) != util.PodScheduled(newPod))
```

Reply:
done, seems to be working

Reply:
nice!